xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -72,10 +72,10 @@ define(["jquery"], function ($) {
     var self = this;
     var defer = $.Deferred();
 
-    layoutId = layoutId || this.data.query.defaultLayout;
-    if (layoutId === this.data.query.currentLayout) return;
-    if (!this.data.meta.layoutDescriptors[layoutId]) return;
-    if (this.data.query.layouts.indexOf(layoutId) === -1) return;
+    layoutId = layoutId || this.data.meta.defaultLayout;
+    if (layoutId === this.data.query.currentLayout) return defer.reject();
```

Comment:
Why not resolve? The layout is basically already loaded.

Reply:
I returned a rejected promise because if we want to execute code after the layout actually changed, in that case nothing happens, and it's still the same layout. But indeed this is not right with the name of the function (**loadlayout**), so I should either I return `defer.resolve()`, or I change the name of the function to `changeLayout`

Reply:
> either return defer.resolve(), or change the name of the function to changeLayout

I would do both :)